mgr/dashboard: Prometheus integration (#25309)
ceph/ceph: doc/mgr/dashboard.rst

  $ ceph dashboard sso enable saml2

Enabling Prometheus alerting
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 

Reviewer: Please add a short note about what the Prometheus Alert manager is all about, including an URL that points to more information, especially how to install and configure it.

> You can use alert manager capabilities within the dashboard in three different

Reviewer 2: ```suggestion
The Alertmanager capabilities can be consumed by the dashboard in three different
```

ways:

#. Use the notification receiver of the dashboard.

#. Use the Prometheus Alertmanager API.

#. Use both sources simultaneously.

> 

Reviewer: What are the benefits/differences between these methods here?
When would one want to use one or the other?
Please elaborate on the rationale for having these two different methods.

> In all methods you will receive alerts. Even in the last method you should only

Reviewer 2: ```suggestion
All three methods are going to provide you with alerts. You will only receive one alert
```

> get one alert, even if both methods want to show alerts.

Reviewer 2: ```suggestion
in the third case, even if both methods are sending alerts to you.
```

#. Use the notification receiver of the dashboard:

   Add the dashboard receiver to your alertmanager configuration and add a

>    route to the new receiver. This should look like::

Reviewer 2: ```suggestion
   This should look like::
```

     route:
       receiver: 'ceph-dashboard'
     ...
     receivers:
       - name: 'ceph-dashboard'
         webhook_configs:
         - url: '<url-to-dashboard>/api/prometheus_receiver'

>    If now the alertmanager sends out an alert notification a pop up will be

Reviewer 2: ```suggestion
   If the Alertmanager sends out an alert notification a pop up containing
```

>    shown on the dashboard showing what alerts have failed.

Reviewer 2: ```suggestion
   the failed alerts will be shown in the dashboard.
```

>    Please make sure that your SSL certificate is valid for the dashboard in the

Reviewer 2: ```suggestion
   Please make sure that the Alertmanager considers your SSL certificate in terms
```

>    eyes of the alertmanager. For more information about the correct

Reviewer 2: ```suggestion
   of the dashboard as valid. For more information about the correct
```

   configuration checkout the `<http_config> documentation
   <https://prometheus.io/docs/alerting/configuration/#%3Chttp_config%3E>`_.

#. To use the API of the Prometheus alertmanager

   To use it, specify the host and port of the alertmanager server::

     $ ceph dashboard set-alertmanager-api-host <alertmanager-host:port>  # default: ''

   For example::

     $ ceph dashboard set-alertmanager-api-host 'http://localhost:9093'

>    After this is set, you will be notified if an alert is changed somehow. You

Reviewer 2: ```suggestion
   After this is set, you will be notified if an alert has been changed somehow. You
```

>    can now see all alerts the alertmanager currently knows of in a tab inside

Reviewer 2: ```suggestion
   can now see all alerts the Alertmanager currently knows of in a tab inside
```

#. Use both sources simultaneously.

All three methods are going to notify you about alerts. You won't be notified
twice if you use both sources.

#. Use the notification receiver of the dashboard:

   This allows you to get notifications as `configured
   <https://prometheus.io/docs/alerting/configuration/>`_ from the alertmanager.
   You will get notified inside the dashboard once a notification is send out,
   but you are not able to manage alerts.

   Add the dashboard receiver and the new route to your Alertmanager configuration.
   This should look like::

     route:
       receiver: 'ceph-dashboard'
     ...
     receivers:
       - name: 'ceph-dashboard'
         webhook_configs:
         - url: '<url-to-dashboard>/api/prometheus_receiver'

   Please make sure that the Alertmanager considers your SSL certificate in terms
   of the dashboard as valid. For more information about the correct
   configuration checkout the `<http_config> documentation
   <https://prometheus.io/docs/alerting/configuration/#%3Chttp_config%3E>`_.

#. Use the API of the Prometheus Alertmanager

   This allows you to manage alerts. You will see all alerts, the Alertmanager

>    currently knows of, in the alters listing. It can be found in the *Cluster*

Reviewer: Small typo:
```suggestion
   currently knows of, in the alerts listing. It can be found in the *Cluster*
```

   submenu as *Alerts*. The alerts in can be sorted by name, job, severity,

>    state and start time. Sadly it's not possible to know when a alert was send

Reviewer: ```suggestion
   state and start time. Unfortunately it's not possible to know when an alert was sent
```
